sdk/src/types/block/unlock/mod.rs

```diff
@@ -38,10 +38,52 @@ pub const UNLOCK_INDEX_RANGE: RangeInclusive<u16> = 0..=UNLOCK_INDEX_MAX; // [0.
 
 pub(crate) type UnlockIndex = BoundedU16<{ *UNLOCK_INDEX_RANGE.start() }, { *UNLOCK_INDEX_RANGE.end() }>;
 
+#[derive(Debug, PartialEq, Eq, derive_more::Display, derive_more::From)]
+#[allow(missing_docs)]
+pub enum UnlockError {
```

Comment:
I know it's a bit annoying but I would put all errors in their own file. We can do that at a later point though, not necessarily in this PR.

Reply:
https://github.com/iotaledger/iota-sdk/issues/2096